### PR TITLE
Add pre and post fields for overriding ToC heading

### PR DESCRIPTION
**Pull request by IveGotNorto**
Pre and post allow users to add semantically correct headings to ToC
without breaking existing functionality.

See #8338

### CODE REVIEW

**Comment by jmooring** [comment]
@IveGotNorto

I pulled this locally for some quick testing. Comments...

- [x] If `pre` is not defined in the site configuration, its value should be `<nav id="TableOfContents">`
- [x] If `post` is not defined in the site configuration, its value should be `</nav>`
- [ ] I should be able to override `pre` independently of `post`. For example:

   ```toml
   [markup]
     [markup.tableOfContents]
       pre = "<nav><h1>Navigation</h1>"
   ```

- [ ] I should be able to override `post` independently of `pre`. For example:

   ```toml
   [markup]
     [markup.tableOfContents]
       post = "</nav><!-- end of nav -->"
   ```

- [ ] If `pre = ""` in the site configuration, the outer `<ol>` or `<ul>` should not be preceded by anything (i.e., do not insert `<nav id="TableOfContents">`).
- [ ] If `post = ""` in the site configuration, the outer `</ol>` or `</ul>` should not be followed by anything (i.e., do not insert `</nav>`).

**Comment by jmooring** [comment]
@IveGotNorto Thank you for working on this!

File: docs/content/en/getting-started/configuration-markup.md [hunk 1]
```diff
@@ -116,6 +116,12 @@ endLevel
 ordered
 : Whether or not to generate an ordered list instead of an unordered list.
 
+pre
+: Value of the pre key if set for overriding the heading entry. This value typically contains a string representing HTML.
```
Comment (jmooring) [inline docs/content/en/getting-started/configuration-markup.md:120]:
```suggestion
: HTML to immediately precede the outer list element. Default value is `<nav id="TableOfContents">`.
```

File: docs/content/en/getting-started/configuration-markup.md [hunk 2]
```diff
@@ -116,6 +116,12 @@ endLevel
...
+
+post
+: Value of the post key if set for overriding the heading entry. This value typically contains a string representing HTML.
```
Comment (jmooring) [inline docs/content/en/getting-started/configuration-markup.md:123]:
```suggestion
: HTML to immediately follow the outer list element. Default value is `</nav>`.
```

**Comment by jmooring** [comment]
@IveGotNorto 

Although this is working as intended, the defaults for pre and post should not be nil. 
<https://github.com/gohugoio/hugo/pull/8643/files#diff-be327cc7b6df56b7b25c4775d78955598913504707bd422132aec74e1b5ffc39R169-R170>

The defaults should be `<nav id="TableOfContents">` and `</nav>`.

**Comment by IveGotNorto** [comment]
> @IveGotNorto
> 
> Although this is working as intended, the defaults for pre and post should not be nil.
> https://github.com/gohugoio/hugo/pull/8643/files#diff-be327cc7b6df56b7b25c4775d78955598913504707bd422132aec74e1b5ffc39R169-R170
> 
> The defaults should be `<nav id="TableOfContents">` and `</nav>`.

I've changed the config defaults for pre and post. The unit tests are passing. How is everything looking now?

**Comment by bep** [comment]
See #8670

**Comment by IveGotNorto** [comment]
Not really sure where this PR stands. Should I be making some different changes related to #8670? @bep @jmooring

**Comment by jmooring** [comment]
I wanted to revist this today to test your changes, but it no longer builds due to https://github.com/gohugoio/hugo/commit/a7e3da242f98d4799dad013d7ba2f285717640d6.

I cannot answer your question about where this PR stands in relation to https://github.com/gohugoio/hugo/issues/8670 and https://github.com/gohugoio/hugo/issues/8680. It seems like we're in a holding pattern.

**Pull request closed without merging**